doc/metadata-server/README.md

```diff
@@ -0,0 +1,327 @@
+## Abstract
+
+This document summarizes requirements and proposes a draft of a protocol for metadata information exchange for the on-chain transactions (`Tx`) which executes Marlowe contracts. The off-chain metadata storage mechanism does not rely on a centralized registry but provides schema which can be implemented by independent metadata providers.
+
+## Motivation
+
+Marlowe is a high-level language with a set of properties which are crucial for writing reliable financial contracts. A possibility to audit the code of the contract which was already initiated on-chain seems to be critical to any participant "invited" to it. Marlowe and its current implementation on Cardano chain gives us access to the source code of the contract, ability to do static analysis over it and "step by step" simulation so it provides the fundamental layer of trust for the language users.
+
+To build a contract exchange platform (Marlowe Marketplace) and execution environment (Marlowe Run) we want to be able to also associate and provide contract template information suitable for human consumption (like title, description, statistics and certification etc.). We want to verify that provided metadata information about the transaction template is consistent with the Marlowe contract embedded in the transaction on-chain. Additionally we should allow the contract initiator to attach a specific set of data (like documents, agreements) to the particular contract instance. We want to be able to provide a validity proof for these data and present it to all other participants as well if necessary.
+
+We want to implement this in an efficient manner avoiding huge payloads on the blockchain and by use of existing or emerging Cardano standards. Additionally we need to keep the protocol decentralized so anybody can take the role of a metadata provider.
```

Comment:
I think this could be much more to the point. Focusing on explaining the reader why we need this. For example:
> When participating on a Marlowe contract only a small amount of information is stored on-chain. Most of the time we can obtain the full code of the contract by looking on-chain, but this is not even true if the contract is merkleized.
> 
> In addition to the actual contract, there are several other pieces of information that would be useful for the participants to know, and without a metadata server there is no standard way for them to get such information.
> 
> Useful information that participants would like to be able to obtain from running contracts include:
> - Role descriptions
> - The template that produced the contract
> - Audit information (i.e: a signed certification from an auditor)
> - etc
> 
Then the issues we face (you did write them, but would be good to add justification):
> A naïve solution would be just add all the information to the chain, but that would be very expensive, bloat the chain, and still it would not be safe, cause someone could forge it. We could just store it in a centralised server, but we want the system to be decentralised to match the nature of the blockchain and to prevent a single point of failure. And it would be desirable to follow existing standards to promote interoperability.

And then an overview of the plan would be good before going into the detail too:

> In order to solve all these problems, this document:
> - Proposes using servers that follow the standard CIP-26
> - Defines a schema for using in conjunction with the standard CIP-26
> - Describe a system to link entries in the schema to contracts on-chain, and a way to verify those links so that entries cannot be forged by bad actors
> - etc